pandas/core/frame.py

```diff
@@ -5813,26 +5814,35 @@ def groupby(
         as_index: bool = True,
         sort: bool = True,
         group_keys: bool = True,
-        squeeze: bool = False,
+        squeeze: bool = no_default,
         observed: bool = False,
     ) -> "DataFrameGroupBy":
         from pandas.core.groupby.generic import DataFrameGroupBy
 
+        if squeeze is not no_default:
+            warnings.warn(
+                (
+                    "The `squeeze` parameter in pd.groupby is deprecated and "
+                    "will be removed in a future version."
+                ),
+                FutureWarning,
+                stacklevel=2,
+            )
+
         if level is None and by is None:
             raise TypeError("You have to supply one of 'by' and 'level'")
         axis = self._get_axis_number(axis)
-
-        return DataFrameGroupBy(
+        grouped = DataFrameGroupBy(
             obj=self,
             keys=by,
             axis=axis,
             level=level,
             as_index=as_index,
             sort=sort,
             group_keys=group_keys,
-            squeeze=squeeze,
```

Comment:
Want to revert this as well; just deprecate for now will remove later for pandas 2.0

Reply:
Yes, I will revert the commits and add only the warning for SeriesGroupBy and DataFrameGroupBy.

Reply:
I reverted the actual code changes. There is only a warning raised right now.